client/app/components/composites/AvatarDropdown/AvatarDropdown.js

```diff
@@ -3,27 +3,54 @@ import r, { div } from 'r-dom';
 import classNames from 'classnames';
 
 import { className } from '../../../utils/PropTypes';
-import { hasTouchEvents } from '../../../utils/featureDetection';
 import ProfileDropdown from './ProfileDropdown';
 import Avatar from '../../elements/Avatar/Avatar';
 import NotificationBadge from '../../elements/NotificationBadge/NotificationBadge';
 
 import css from './AvatarDropdown.css';
 
+const HOVER_TIMEOUT = 250;
+
 class AvatarDropdown extends Component {
   constructor(props, context) {
     super(props, context);
 
+    this.handleMouseOver = this.handleMouseOver.bind(this);
+    this.handleMouseOut = this.handleMouseOut.bind(this);
     this.handleClick = this.handleClick.bind(this);
     this.handleBlur = this.handleBlur.bind(this);
 
     this.state = {
       isOpen: false,
+      isHovering: false,
+      isMounted: false,
     };
+
+    this.mouseOverTimeout = null;
+    this.mouseOutTimeout = null;
+  }
+
+  componentDidMount() {
+    this.setState({ isMounted: true });// eslint-disable-line react/no-set-state
+  }
+
+  handleMouseOver() {
+    clearTimeout(this.mouseOutTimeout);
+    clearTimeout(this.mouseOverTimeout);
```

Comment:
Let's prefix all these native global functions with `window`, ok? We should probably turn on the ESLint rule to complain about those as well.